recipes/imagemagick/all/conanfile.py

```diff
@@ -61,375 +75,321 @@ class ImageMagicConan(ConanFile):
-            "MD": "DynamicMT",
-            "MDd": "DynamicMT",
-        }.get(str(self.settings.compiler.runtime))
-
-        # GdiPlus requires C++, but ImageMagick has *.c files
-        project = (
-            "IM_MOD_emf_%s.vcxproj" % suffix
-            if self.options.shared
-            else "CORE_coders_%s.vcxproj" % suffix
-        )
-        tools.replace_in_file(
-            os.path.join("VisualMagick", "coders", project),
-            '<ClCompile Include="..\\..\\ImageMagick\\coders\\emf.c">',
-            '<ClCompile Include="..\\..\\ImageMagick\\coders\\emf.c">\n'
-            "<CompileAs>CompileAsCpp</CompileAs>",
-        )
-
-        for module in self._modules:
-            with tools.chdir(os.path.join("VisualMagick", module)):
-                msbuild = MSBuild(self)
-                msbuild.build(
-                    project_file="CORE_%s_%s.vcxproj" % (module, suffix),
-                    upgrade_project=False,
-                    platforms={"x86": "Win32", "x86_64": "x64"},
-                )
-
-        with tools.chdir(os.path.join("VisualMagick", "coders")):
-            pattern = (
-                "IM_MOD_*_%s.vcxproj" % suffix
-                if self.options.shared
-                else "CORE_coders_%s.vcxproj" % suffix
-            )
-            projects = glob.glob(pattern)
-            for project in projects:
-                msbuild = MSBuild(self)
-                msbuild.build(
-                    project_file=project,
-                    upgrade_project=False,
-                    platforms={"x86": "Win32", "x86_64": "x64"},
-                )
+        files.apply_conandata_patches(self)
+
+        if microsoft.is_msvc(self):
+            # jpeg library is named as libjpeg in Windows
+            files.replace_in_file(self, os.path.join(
+                self.source_folder, "configure"), "-ljpeg", "-llibjpeg")
+
+            # AutotoolsDeps makes cl include directories of all dependencies and
+            # link against all dependencies, which is useful since this is the only
+            # way for imagemagick to find libjpeg for example
+            # but one of these dependencies defines the "select" function, which
+            # makes imagemagick incorrectly think that Windows supports the POSIX
+            # select function
+            files.replace_in_file(self, os.path.join(self.source_folder, "configure"),
+                                  "#define HAVE_SELECT 1", "/* #undef HAVE_SELECT */")
+
+            self.conf["tools.microsoft.bash:subsystem"] = "msys2"
+
+        # FIXME: change pangocairo pkg-config component name in the pango recipe
+        if os.path.exists(os.path.join(self.generators_folder, "pango-pangocairo.pc")):
+            files.rename(
+                self,
+                os.path.join(self.generators_folder, "pango-pangocairo.pc"),
+                os.path.join(self.generators_folder, "pangocairo.pc"))
+
+        autotools = self._build_configure()
```

Comment:
```suggestion
        autotools = Autotools()
        autotools.configure()
```
DO not use _configure_xxx when using new helpers.